tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2260,14 +2310,19 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
               "Invalid tree in JSDesugar.transformExpr() of class " +
               tree.getClass)
       }
+
+      if (preserveChar || tree.tpe != CharType)
+        baseResult
+      else
+        genCallHelper("bC", baseResult)
```

Comment:
Hum, doesn't this unbox too often? For example, if you have a `Block(..., x)` of type `CharType`. When you transform `x` you already unbox there (since you transform it with `preserveChar = false`, but then you also unbox the resulting block. I think you need to follow the expected types of the original trees :-/